pkg/permissions/generate.go

```diff
@@ -113,3 +188,17 @@ func GenerateAgentRBACManifests(dataGatherers []agent.DataGatherer) AgentRBACMan
 
 	return AgentRBACManifests
 }
+
+func GenerateFullManifest(dataGatherers []agent.DataGatherer) string {
+	agentRBACManifestsStruct := GenerateAgentRBACManifests(dataGatherers)
+	agentCLR := AgentCLR(agentRBACManifestsStruct.ClusterRoles)
+	agentCLRB := AgentCLRB(agentRBACManifestsStruct.ClusterRoleBindings)
+	agentRB := AgentRB(agentRBACManifestsStruct.RoleBindings)
+
+	out := fmt.Sprintf(
+		`%s
+%s
+%s`, agentCLR, agentCLRB, agentRB)
```

Comment:
in this way that you have here of concatenating strings, you are not considering when either `agentCLR,agentCLRB` or `agentRB` are empty strings.

example if `agentCLRB` is an empty string:
```
        apiVersion: rbac.authorization.k8s.io/v1
        kind: ClusterRole
        metadata:
          name: jetstack-secure-agent-pods-reader
        rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get", "list", "watch"]
    --> Here you're going to have a new line for the missing  `agentCLRB`    
        ---
        apiVersion: rbac.authorization.k8s.io/v1
        kind: RoleBinding
        metadata:
          name: jetstack-secure-agent-pods-reader
          namespace: foobar
        roleRef:
          kind: ClusterRole
          name: jetstack-secure-agent-pods-reader
          apiGroup: rbac.authorization.k8s.io
        subjects:
        - kind: ServiceAccount
          name: agent
          namespace: jetstack-secure
``

Reply:
a solution could check if a variable is an empty string before appending it to the output